drivers/dma/dma_ti_cc23x0.c

```diff
@@ -0,0 +1,380 @@
+ * assert a DMA request signal whenever the peripheral is ready for a data transfer.
+ * Auto mode is similar to basic mode, except that when a transfer request is received,
+ * the transfer completes, even if the DMA request is removed. This mode is suitable for
+ * software-triggered transfers.
+ */
+#define DMA_CC23_MODE(ch)	(DMA_CC23_IS_SW_CH(ch) ? UDMA_MODE_AUTO : UDMA_MODE_BASIC)
+
+/* Each DMA channel is multiplexed between two peripherals which ID is in range 0 to 7 */
+#define DMA_CC23_IPID_MASK	GENMASK(2, 0)
+#define DMA_CC23_CHXSEL_REG(ch)	HWREG(EVTSVT_BASE + EVTSVT_O_DMACH0SEL + sizeof(uint32_t) * (ch))
+
+struct dma_cc23x0_channel {
+	uint8_t data_size;
+	dma_callback_t cb;
+	void *user_data;
+};
+
+struct dma_cc23x0_data {
+	__aligned(1024) uDMAControlTableEntry desc[UDMA_NUM_CHANNELS];
+	struct dma_cc23x0_channel channels[UDMA_NUM_CHANNELS];
+};
+
+/*
+ * If the channel is a software channel, then the completion will be signaled
+ * on this DMA dedicated interrupt.
+ * If a peripheral channel is used, then the completion will be signaled on the
+ * peripheral's interrupt.
+ */
+static void dma_cc23x0_isr(const struct device *dev)
+{
+	struct dma_cc23x0_data *data = dev->data;
+	struct dma_cc23x0_channel *ch_data;
+	uint32_t done_flags;
+	int i;
+
+	done_flags = uDMAIntStatus();
+
+	for (i = DMA_CC23_SW_CH_MIN ; i < DMA_CC23_SW_CH_MAX ; i++) {
+		if (done_flags & BIT(i)) {
+			LOG_DBG("DMA transfer completed on channel %d", i);
+
+			ch_data = &data->channels[i];
+			if (ch_data->cb) {
+				ch_data->cb(dev, ch_data->user_data, i, DMA_STATUS_COMPLETE);
+			}
+
+			uDMAClearInt(done_flags & BIT(i));
+		}
+	}
+}
+
+static uint32_t dma_cc23x0_set_addr_adj(uint32_t *control, uint16_t addr_adj,
+					uint32_t inc_flags, uint32_t no_inc_flags)
+{
+	switch (addr_adj) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*control |= inc_flags;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*control |= no_inc_flags;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_cc23x0_config(const struct device *dev, uint32_t channel,
+			     struct dma_config *config)
+{
+	struct dma_cc23x0_data *data = dev->data;
+	struct dma_cc23x0_channel *ch_data;
+	struct dma_block_config *block = config->head_block;
+	uint32_t control;
+	uint32_t data_size;
+	uint32_t src_inc_flags;
+	uint32_t dst_inc_flags;
+	uint32_t xfer_size;
+	uint32_t burst_len;
+	int ret;
+
+	if (channel >= UDMA_NUM_CHANNELS) {
+		LOG_ERR("Invalid channel");
+		return -EINVAL;
+	}
+
+	if (config->dma_slot > DMA_CC23_IPID_MASK) {
+		LOG_ERR("Invalid trigger");
+		return -EINVAL;
+	}
+
+	if (config->block_count > 1) {
+		LOG_ERR("Chained transfers not supported");
+		return -ENOTSUP;
+	}
+
+	switch (config->source_data_size) {
+	case 1:
+		src_inc_flags = UDMA_SRC_INC_8;
+		break;
+	case 2:
+		src_inc_flags = UDMA_SRC_INC_16;
+		break;
+	case 4:
+		src_inc_flags = UDMA_SRC_INC_32;
+		break;
+	default:
+		LOG_ERR("Invalid source data size");
```

Comment:
Consider logging the actual wrong value in the error messages, just adds a couple bytes to the string (if errors are enabled even) and  makes debugging easier, same for errors in `ret < 0` paths